vault.go

```diff
@@ -104,8 +105,10 @@ func addVault(pod *corev1.Pod, namespace string, databases []database) (patch []
 
 		initContainer := vaultContainer
 
+		jobLikeOwnerReferencesKinds := []string{"Job", "Workflow"}
+
 		if len(pod.ObjectMeta.OwnerReferences) != 0 {
-			if pod.ObjectMeta.OwnerReferences[0].Kind == "Job" {
+			if slices.Contains(jobLikeOwnerReferencesKinds, pod.ObjectMeta.OwnerReferences[0].Kind) {
```

Comment:
This makes sense but I'm a little concerned with pulling in code from Golang's `exp` package. That's essentially experimental code that might make it into the standard libs in the future.

I would suggest the following as an option if you don't want to iterate:

```suggestion
		jobLikeOwnerReferencesKinds := map[string]bool{Job: true, Workflow: true}

		if len(pod.ObjectMeta.OwnerReferences) != 0 {
			ownerKind := pod.ObjectMeta.OwnerReferences[0].Kind
			if _, ok := jobLikeOwnerReferencesKinds[ownerKind]; ok {
```